Zainabfadeyi/person-finance-manager
# Implement Sidebar Component and Setup Routing

# Implement Sidebar Component and Setup Routing

## Description
This PR introduces a new Sidebar component and sets up routing for our React application. The Sidebar is responsive, can be toggled open and closed, and highlights the active page. We've also implemented basic routing to support navigation between different pages of the application.

## Changes
- Created a new Sidebar component
- Implemented custom CSS for the Sidebar using CSS modules
- Updated the App component to include routing
- Set up basic page components for each route

## New Components
- `Sidebar.js`: A responsive sidebar with toggle functionality
- `Sidebar.module.css`: Custom styles for the Sidebar component
- Placeholder components for Dashboard, and Transactions pages

## Modified Components
- `App.js`: Updated to include routing and the Sidebar component
- `App.module.css`: Added styles to accommodate the Sidebar

## Features
- Responsive sidebar that can be toggled open and closed
- Active page highlighting in the sidebar
- Icons remain visible when the sidebar is minimized
- Basic routing setup using React Router

## How to Test
1. Run the application and verify that the Sidebar appears on the left side of the screen.
2. Click the toggle button to open and close the Sidebar.
3. Navigate between different pages by clicking on the Sidebar items.
4. Verify that the active page is highlighted in the Sidebar.
5. Minimize the Sidebar and check that icons are still visible.
6. Test on different screen sizes to ensure responsiveness.

## Next Steps
- Implement actual page components
- Add content to each page
- Implement user authentication and protected routes
- Add unit tests for the Sidebar component and routing

Please review and let me know if any changes or additional information is needed.
